### PR TITLE
Happy-thoughts-api-Ivett

### PR DESCRIPTION
**Pull request by Lilivet**
Live project back-end:

https://ivett-happy-thoughts.herokuapp.com/
End-points:
https://ivett-happy-thoughts.herokuapp.com/thoughts  (20 of the newest messages with sorted by date and decreasing)
https://ivett-happy-thoughts.herokuapp.com/thoughtId/like
https://ivett-happy-thoughts.herokuapp.com/id

Live project front-end (updated with new back-end (API urls):
https://happy-thoughts-ivett.netlify.com/

I thought that my mongodb was connected,but it is not yet connected!

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -23,6 +55,61 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// get(read) all the messages posted by the client into the database,
+// sort them by date in a descendent way and limit them to 20
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' })
```
Comment (JennieDalgren) [inline server.js]: ⭐️

File: server.js [hunk 2]
```diff
@@ -23,6 +55,61 @@ app.get('/', (req, res) => {
...
+// 1) Get the message that you want to update (by id)
+app.get('/thoughts/:id', (req, res) => {
+  Thought.findOne({ id: req.params._id }).then(thought => {
+    if (thought) {
+      res.json(thought)
+    } else {
+      res.status(404).json({ error: 'Not found', error: err.errors })
+    }
+  })
+})
```
Comment (JennieDalgren) [inline server.js]: What are you doing with this endpoint? Do you need it? Are you calling this from the frontend at some point?

File: server.js [hunk 3]
```diff
@@ -23,6 +55,61 @@ app.get('/', (req, res) => {
...
+
+// 2) Change the part of the message that you want to update,
+//in this case update the likes (increase # of hearts)
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  // Retrieve the information sent by the client to our API endpoint
+  // in this case is the id of the thought
+  const { thoughtId } = req.params
+  // comparing the id received with our db
+  const thoughtLiked = await Thought.findById(thoughtId)
```
Comment (JennieDalgren) [inline server.js]: maybe wrap this in a try catch so that it's not stuck here at the await if something goes wrong at this point.

**Comment by Lilivet** [comment]
Yes!!! I sent a comment in stackoverflow and I got help from Michel. I
needed to populate with some thoughts the Mongodb. I did it with Postman
and then it started to work!

It that the best way?

I have another comment. I see that the Website is downloading the existing
thoughts (the 20 thought list) very slow. The first thing that you see when
you get in is the Form for new thoughts, but you do not see the existing
ones right away. Is there any way that I can improve this behaviour?

Thanks Jenny!

/Ivett

**Comment by Lilivet** [comment]
On Mon, Feb 17, 2020 at 11:55 AM Jennie Dalgren <notifications@github.com>
wrote:

> *@JennieDalgren* approved this pull request.
>
> Good job with the project!
>
> You wrote in the comments that your database wasn't connected? It looks
> like it's working now, right?
> ------------------------------
>
> In server.js
> <https://github.com/Technigo/project-happy-thoughts-api/pull/3#discussion_r380111512>
> :
>
> > @@ -23,6 +55,61 @@ app.get('/', (req, res) => {
>
>    res.send('Hello world')
>
>  })
>
>
>
> +// get(read) all the messages posted by the client into the database,
>
> +// sort them by date in a descendent way and limit them to 20
>
> +app.get('/thoughts', async (req, res) => {
>
> +  const thoughts = await Thought.find().sort({ createdAt: 'desc' })
>
>
> ⭐️
> ------------------------------
>
> In server.js
> <https://github.com/Technigo/project-happy-thoughts-api/pull/3#discussion_r380112368>
> :
>
> > +app.get('/thoughts/:id', (req, res) => {
>
> +  Thought.findOne({ id: req.params._id }).then(thought => {
>
> +    if (thought) {
>
> +      res.json(thought)
>
> +    } else {
>
> +      res.status(404).json({ error: 'Not found', error: err.errors })
>
> +    }
>
> +  })
>
> +})
>
>
> What are you doing with this endpoint? Do you need it? Are you calling
> this from the frontend at some point?
>
I am not using this endpoint. I just created it as a reference to check the
id of the messages, but I could remove it without affecting the front end.

> ------------------------------
>
> In server.js
> <https://github.com/Technigo/project-happy-thoughts-api/pull/3#discussion_r380112876>
> :
>
> > +    if (thought) {
>
> +      res.json(thought)
>
> +    } else {
>
> +      res.status(404).json({ error: 'Not found', error: err.errors })
>
> +    }
>
> +  })
>
> +})
>
> +
>
> +// 2) Change the part of the message that you want to update,
>
> +//in this case update the likes (increase # of hearts)
>
> +app.post('/thoughts/:thoughtId/like', async (req, res) => {
>
> +  // Retrieve the information sent by the client to our API endpoint
>
> +  // in this case is the id of the thought
>
> +  const { thoughtId } = req.params
>
> +  // comparing the id received with our db
>
> +  const thoughtLiked = await Thought.findById(thoughtId)
>
>
> maybe wrap this in a try catch so that it's not stuck here at the await if
> something goes wrong at this point.
>
Maybe this step is the one that it is stopping the download of existing
messages when you are get into the website for the first time?

**Pull request closed without merging**